app/forms/product_form.rb

```diff
@@ -60,30 +62,52 @@ def attributes(product)
   def save
     return false if invalid?
 
-    product.save! # product.mediaのアソシエーションごと保存される
-    assign_attributes(id: product.id)
+    save!
     true
   end
 
   def update(params)
     assign_attributes(params)
     return false if invalid?
 
-    product.update!(**product_params)
+    product.assign_attributes(**product_params)
+    save!
+    true
   end
 
   def to_model
     product
   end
 
   def media
-    media_attributes.map do |attribute|
-      product.media.build(attribute)
+    media_attributes.map(&:deep_symbolize_keys).map do |attributes|
+      if attributes[:id].present?
+        medium = product.media.find(attributes[:id])
+        medium.assign_attributes(**attributes.slice(:title, :url))
+        medium
+      else
+        # product.mediaでは取得できないようにする
+        ProductMedium.new(**attributes, product: product)
```

Comment:
```suggestion
        product.media.build(**attributes)
```

にすると、
- `product.media`として関連で紐付いたmedium
- `media`として配列で渡されたmedium

が重複して作成されてしまうため
（`product.save`でproduct.mediaも一緒に保存されてしまう）